vignettes/testing-cvd.Rmd

```diff
@@ -0,0 +1,184 @@
+library(ggplot2)
+```
+
+# Background
+
+We are testing that the existing NMFS color palettes are usable to someone with color vision deficiency (C.V.D.).
+
+We'll use an R package called [`dichromat`](https://cran.r-project.org/web/packages/dichromat/index.html) to approximate how these palettes might be perceived by someone with any of the most common forms of C.V.D.: protanopia, deutanopia, and tritanopia.
+
+This analysis is based on [this tutorial by Hannah Weller](https://hiweller.rbind.io/post/using-the-dichromat-package-to-check-if-your-plot-is-colorblind-friendly/).
+
+## Install packages
+```{r eval=FALSE}
+install.packages("dichromat")
+install.packages("recolorize")
+install.packages("gridExtra")
+```
+
+## Load libraries
+```{r}
+library(dichromat)
+library(recolorize)
+library(gridExtra)
+```
+
+## Create a function that shows perceptions of palettes
+```{r}
+show_cvd_pals <- function(palette){
+  # Convert palette into three palettes approximating how someone with each type of C.V.D. would perceive the original palette
+
+  ## protanopia
+  protan <- dichromat::dichromat(palette, type = "protan")
+  ## deutanopia
+  deutan <- dichromat::dichromat(palette, type = "deutan")
+  ## tritanopia
+  tritan <- dichromat::dichromat(palette, type = "tritan")
+  
+  # Plot the palettes and compare
+  layout(matrix(1:4, nrow = 4))
+  par(mar = rep(1, 4))
+  
+  recolorize::plotColorPalette(palette, main = "Trichromacy ('normal color vision')")
+  recolorize::plotColorPalette(protan, main = "Protanopia")
+  recolorize::plotColorPalette(deutan, main = "Deutanopia")
+  recolorize::plotColorPalette(tritan, main = "Tritanopia")
+}
+```
+
+# Test main palettes
+```{r fig.align='center'}
+# import main palettes from nmfs_cols.R file
+core_nmfs_pals <- list(
+  `oceans` = nmfs_cols("pale_sea_blue", "light_sea_blue", "processblue", "reflexblue", "national", "o50pblack"),
+  `waves` = nmfs_cols("pale_teal", "bright_teal", "westcoast", "dark_teal_green"),
+  `seagrass` = nmfs_cols("bright_seagrass", "ltgreen", "southeast", "dark_seagrass"),
+  `urchin` = nmfs_cols("bright_urchin", "vivid_urchin", "midatlantic", "dark_urchin"),
+  `crustacean` = nmfs_cols("bright_crustacean", "ltorange", "alaska", "dark_crustacean"),
+  `coral` = nmfs_cols("bright_coral", "vivid_coral", "pacificislands", "dark_coral"),
+  `regional` = nmfs_cols("national", "westcoast", "southeast", "midatlantic", "alaska", "pacificislands")
+)
+
+# Show comparisons for each palette
+for (i in 1:length(core_nmfs_pals)){
+  print(paste0(names(core_nmfs_pals)[i], " palette"))
+  show_cvd_pals(core_nmfs_pals[[i]])
+}
+```
+
+**Conclusion**: For someone with C.V.D., the regional palette could be challenging to discern. In particular, colors look very similar when viewed with the following perceptions: protanopia (3, 5); deutanopia (3, 5, 6); tritanopia (2, 3, 4).
+
+# Test alternative regional color palette
+## Create alternative palettes
+```{r}
+# Option 1: using existing NMFS colors
```

Comment:
should this instead be code to get the palettes within nmfspalette, rather than code that manually constructs new palettes?